### PR TITLE
Eslint upgraded and fixed

### PR DESCRIPTION
**Pull request by AlteaDown**
## This should only be merged to master after the Alpha release is completed, and we are in the cleanup/bugfixing stage

Heavily revamped ESLint. There were a bunch of issues, almost all the rules were disabled due to mistakes in how ESLint + vue + Typescript configs work together. That means there is a lot to clean up in this project. I've set all the rules up, such that they match the best practices for TS and for Vue. The project will light up like a christmas tree until a ESLint auto fix is run on the project.

Also realized there is more than 1 eslintrc file, and that we were trying to programmatically override the root eslint config, but that's already a feature in ESLint, so I instead used the first class method of doing this, so that IDE's can parse the files correctly.

I will be going over this branch in detail, during the upcoming frontend-chapter meetup.

### CODE REVIEW

File: .eslintrc.js
Comment (safydy-r) [inline .eslintrc.js:210]: Thanks for thinking of this, ;)

File: .editorconfig
Comment (safydy-r) [inline .editorconfig:1]: Thanks for updating this

Comment (AlteaDown) [reply .editorconfig:1]: These can actually be generated from IntelliJ, so that's exactly what I did. I made one that matches our standards, and exported it as `.editorconfig`

File: .eslintrc.js
Comment (safydy-r) [inline .eslintrc.js:135]: I don't know if this is caused by this rule:
this line https://github.com/FLOIP/flow-builder/blob/eslint-upgraded-and-fixed/src/components/interaction-designer/flow-editors/ModesEditor.vue#L36 gives us an error

```
ERROR in /Users/rs/Documents/Voto/floip/flow-builder/src/components/interaction-designer/flow-editors/ModesEditor.vue(35,13):
35:13 'availableModes' is declared but its value is never read.
    33 |     @Prop() readonly flow!: IFlow
    34 |
  > 35 |     private availableModes = Object.values(SupportedMode)
       |             ^
    36 |
    37 |     get flowSelectedModes(): SupportedMode[] {
    38 |       return this.flow.supported_modes
```
    
While `availableModes` is used in template, any idea how to fix it?

**Review by safydy-r — CHANGES_REQUESTED** [review]
Thanks @AlteaDown for spending time on this, overall I'm excited to have these merged.
I left questions & notes for you.

And FYI, I made a commit to clean some unused vars.

File: src/components/common/PlainDraggable.vue
Comment (AlteaDown) [reply src/components/common/PlainDraggable.vue:85]: For these cases, use block comments then /* */ as they are made for these cases. That way, when someone is maintaining this later, they know what comments were related intentionally, including multiple lines of code.

File: src/components/interaction-designer/BuilderCanvas.vue
Comment (AlteaDown) [reply src/components/interaction-designer/BuilderCanvas.vue]: saw this and fixed them.

File: src/components/interaction-designer/BuilderCanvas.vue
Comment (AlteaDown) [reply src/components/interaction-designer/BuilderCanvas.vue]: fixed these too.

File: src/components/interaction-designer/resource-editors/ResourceEditor.vue
Comment (AlteaDown) [reply src/components/interaction-designer/resource-editors/ResourceEditor.vue:53]: Right now, the rule is if you have 1 attribute, it can be on 1 line. any more than that, and line breaks are needed. When in doubt, run code cleanup command in PHPStorm, then do ESLint autofix, and it will put it in the correct format. Technically all modified files should now have code cleanup command in PHPStorm, then ESLint autofix on all changed files before committing changes (this can be automated to happen on save of files in PHPStorm). This way, the code format is always correct, and we never add additional tech debt due to mixed code styles.

File: src/store/trees/bootstrap-legacy-global-dependencies.js
Comment (AlteaDown) [reply src/store/trees/bootstrap-legacy-global-dependencies.js:4]: yes, this is now part of the code style in the `.editorconfig` as well as due to ESLint rules.

File: src/views/NewFlow.vue
Comment (AlteaDown) [reply src/views/NewFlow.vue:15]: This is because the vue was actually invalid, (we passed a param without quotes, which is not valid)

File: stories/story-utils/storeSetup.ts
Comment (AlteaDown) [reply stories/story-utils/storeSetup.ts:125]: I put it back in.

File: .eslintrc.js
Comment (AlteaDown) [reply .eslintrc.js:135]: I turned off this rule in tsconfig, so it' should be good now.

File: stories/story-utils/storeSetup.ts
Comment (safydy-r) [reply stories/story-utils/storeSetup.ts:125]: oh thanks, I didn't know we can do this
```
// @ts-ignore
content_type: [SupportedContentType.TEXT],
```

**Review by safydy-r — APPROVED** [review]
Thanks you @AlteaDown , this was a huge one.

I'd like to wait for at least one other dev to approve before we merge.
cc: @poojahpatil90 , @jcbashdown

**Review by poojahpatil90 — APPROVED** [review]
Overall looks good. Left a few comments.

File: .eslintrc.js
Comment (AlteaDown) [reply .eslintrc.js:59]: This makes it so we can disable the rules that would definitely break on `.js` files. We disable typescript rules for JS files, because if you want to migrate a JS file to TS, you should change the extension to `.ts`.  If you want to start getting rules for TS, change the extension from `.js`, to `.ts` and you will start getting them. This can be done because, JS is valid TS, but TS is not valid JS.

File: src/components/resource-editor/HorizontalBlockContentEditor.vue
Comment (AlteaDown) [reply src/components/resource-editor/HorizontalBlockContentEditor.vue:142]: Fixed.

**Pull request merged**
